group-coordinator/src/main/java/org/apache/kafka/coordinator/group/RecordHelpers.java

```diff
@@ -0,0 +1,371 @@
+import org.apache.kafka.coordinator.group.generated.ConsumerGroupTargetAssignmentMetadataValue;
+import org.apache.kafka.server.common.ApiMessageAndVersion;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class contains helper methods to create records stored in
+ * the __consumer_offsets topic.
+ */
+public class RecordHelpers {
+    private RecordHelpers() {}
+
+    /**
+     * Creates a ConsumerGroupMemberMetadata record.
+     *
+     * @param groupId   The consumer group id.
+     * @param member    The consumer group member.
+     * @return The record.
+     */
+    public static Record newMemberSubscriptionRecord(
+        String groupId,
+        ConsumerGroupMember member
+    ) {
+        return new Record(
+            new ApiMessageAndVersion(
+                new ConsumerGroupMemberMetadataKey()
+                    .setGroupId(groupId)
+                    .setMemberId(member.memberId()),
+                (short) 5
+            ),
+            new ApiMessageAndVersion(
+                new ConsumerGroupMemberMetadataValue()
+                    .setRackId(member.rackId())
+                    .setInstanceId(member.instanceId())
+                    .setClientId(member.clientId())
+                    .setClientHost(member.clientHost())
+                    .setSubscribedTopicNames(member.subscribedTopicNames())
+                    .setSubscribedTopicRegex(member.subscribedTopicRegex())
+                    .setServerAssignor(member.serverAssignorName().orElse(null))
+                    .setRebalanceTimeoutMs(member.rebalanceTimeoutMs())
+                    .setAssignors(member.clientAssignors().stream().map(assignorState ->
+                        new ConsumerGroupMemberMetadataValue.Assignor()
+                            .setName(assignorState.name())
+                            .setReason(assignorState.reason())
+                            .setMinimumVersion(assignorState.minimumVersion())
+                            .setMaximumVersion(assignorState.maximumVersion())
+                            .setVersion(assignorState.metadata().version())
+                            .setMetadata(assignorState.metadata().metadata().array())
+                    ).collect(Collectors.toList())),
+                (short) 0
+            )
+        );
+    }
+
+    /**
+     * Creates a ConsumerGroupMemberMetadata tombstone.
+     *
+     * @param groupId   The consumer group id.
+     * @param memberId  The consumer group member id.
+     * @return The record.
+     */
+    public static Record newMemberSubscriptionTombstoneRecord(
+        String groupId,
+        String memberId
+    ) {
+        return new Record(
+            new ApiMessageAndVersion(
+                new ConsumerGroupMemberMetadataKey()
+                    .setGroupId(groupId)
+                    .setMemberId(memberId),
+                (short) 5
```

Comment:
Are we using version here to distinguish what the type of the key is? Sorry if I missed this somewhere.

Reply:
Or is this the versions that we are on and we are just hard coding them here.

Reply:
That is correct. The version is used to identify the key type here.

Reply:
Interesting -- do other usages of this class do this? It seems a little confusing that the field is called version.

Reply:
I agree that it is confusing. I was thinking about not using `ApiMessageAndVersion` but I would keep this for a later refactoring.

Reply:
Opened https://issues.apache.org/jira/browse/KAFKA-14940.